docs/tutorials/03_authenticated_user.md

```diff
@@ -0,0 +1,98 @@
+
+- [Forsta Console]{@link https://app.forsta.io/people}
+- [Atlas API]{@link https://atlas.forsta.io/doc/}
+
+For a user to authenticate with the Forsta Messenger Client they need to provide
+a Forsta generated JWT (JSON Web Token). In a production system JWTs are generated
+by authenticating a user with a saved auth token. For testing purposes you can
+also generate a JWT with the following shell scripts:
+
+- [JWT Generator]{@link https://github.com/ForstaLabs/developer-examples/blob/master/scripts/get_jwt.sh}
+- [Required Forsta Script Library]{@link https://github.com/ForstaLabs/developer-examples/blob/master/scripts/forsta_library.sh}
+
+Once you have generated a jwt start with the following code that authenticates a user
+in the Forsta Messenger Client:
+
+```html
+<html>
+  <head>
+    <script src="https://forstalabs.github.io/forsta-messenger-client/dist/forsta-messenger-client.min.js"></script>
+  </head>
+
+  <body>
+    <!-- Div that contains the Forsta messenger client -->
+    <div id="my-messenger" style="width: 80%; height: 80%; visibility: hidden;"></div>
+  </body>
+
+  <script>
+    function displayMessenger() {
+      const messenger = document.getElementById("my-messenger");
+      messenger.style.visibility = "visible";
+    }
+    
+    async function onLoaded(client) {
+      threadId = await client.threadMake("@userinyourorg");
+      await client.threadOpen(threadId);
+      
+      displayMessenger();
+    }
+
+    // Configure the Forsta messenger client to use the div with id my-messenger
+    // And connect as an ephemeral user using the ephemeral chat token
+    const myClient = new forsta.messenger.Client(document.getElementById('my-messenger'),
+      { jwt: 'server generated jwt' },
+      { onLoaded: onLoaded, showNav: true, showHeader: true });
+  </script>
+</html>
+
+```
+
+If you load this code for the first time in a browser you will discover that nothing displays.
+This is because authenticated users on new systems must first provision their encryption keys.
+The `onLoaded` function fires after provisioning. To display the messenger when provisioning is
+required there are three provisioning listeners. Add the following functions to your script:
+
+```html
+function onProvisioningRequired() {
+  displayMessenger();
+}
+
+function onProvisioningError() {
+  alert("Provisioning Failed!");
+}
+
+function onProvisioningDone() {
+  alert("Provisioning Finished!");
+}
+```
+
+Then after the `myClient =` line add the following to set the event listeners:
+
+```html
+myClient.addEventListener('provisioningrequired', onProvisioningRequired);
+myClient.addEventListener('provisioningerror', onProvisioningError);
+myClient.addEventListener('provisioningdone', onProvisioningDone);
+```
+
+Now, try loading the page. The provisioning listener will trigger the messenger to display.
+You have the option to either:
+
+- Generate new keys
+- Import previous keys from a currently running client
```

Comment:
I think (but am not 100%) all the language is singular as it refers to the import or generation of THE identity key.  There is just one.

Reply:
confirmed:
![image](https://user-images.githubusercontent.com/139316/67311541-04eabe00-f4bd-11e9-9b20-280f260658d4.png)

Reply:
Fixed